src/Controls/tests/TestCases.HostApp/Issues/Issue21980.xaml

```diff
@@ -0,0 +1,80 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             x:Class="Maui.Controls.Sample.Issues.Issue21980">
+    <ContentPage.Resources>
+        <Style x:Key="IndicatorViewTemplateStyle" TargetType="Image">
+            <Setter Property="VisualStateManager.VisualStateGroups">
+                <VisualStateGroupList>
+                    <VisualStateGroup x:Name="CommonStates">
+                        <VisualState x:Name="Normal">
+                            <VisualState.Setters>
+                                <Setter Property="Source" Value="indicator_unselected" />
+                            </VisualState.Setters>
+                        </VisualState>
+                        <VisualState x:Name="Selected">
+                            <VisualState.Setters>
+                                <Setter Property="Source" Value="indicator_selected" />
+                            </VisualState.Setters>
+                        </VisualState>
+                    </VisualStateGroup>
+                </VisualStateGroupList>
+            </Setter>
+        </Style>
+    </ContentPage.Resources>
+        <VerticalStackLayout>
+            <Grid>
+                <CarouselView
+                    ItemsSource="{Binding Images}"
+                    IndicatorView="IndicatorView"
+                    HorizontalScrollBarVisibility="Never"
+                    BackgroundColor="#ececec"
+                    WidthRequest="300" 
+                    HeightRequest="450"
+                    Grid.Row="0">
+                    <CarouselView.ItemTemplate>
+                        <DataTemplate>
+                            <Grid>
+                                <Image 
+                                    Source="{Binding .}" 
+                                    WidthRequest="300" 
+                                    HeightRequest="450"/>
+                            </Grid>
+                        </DataTemplate>
+                    </CarouselView.ItemTemplate>
+                </CarouselView>
+                <Border
+                    HorizontalOptions="Center"
+                    VerticalOptions="End"
+                    Margin="16"
+                    Padding="3"
+                    BackgroundColor="White"
+                    StrokeShape="RoundRectangle 12"
+                    StrokeThickness="0"
+                    Grid.Row="0">
+                    <IndicatorView
```

Comment:
The IndicatorView behavior seems wrong on Android, it is not sync with the CV position.

Reply:
I've also stumbled upon this, but I think it is an unrelated pre-existing bug that happens only when image sources are the same. 

|The same images|Different images|
|--|--|
|<video src="https://github.com/user-attachments/assets/0d092720-3380-4642-b917-00495e7c4666" width="300px"/>|<video src="https://github.com/user-attachments/assets/3f202d79-6d80-4572-aa1b-d30c21d7d5fc" width="300px"/>|